Core/CoreTests/Search/CoreSearchHostingControllerTests.swift

```diff
@@ -117,6 +117,8 @@ class CoreSearchHostingControllerTests: CoreTestCase {
         textField.text = "Example"
         _ = textField.delegate?.textFieldShouldReturn?(textField)
 
+        drainMainQueue(thoroughness: 10)
+
```

Comment:
This doesn't fix the flaky test issue, the test still fails on this PR. I added a workaround in a dedicated test fix PR, please take a look.

Reply:
<img width="792" alt="Screenshot 2025-01-08 at 11 04 48" src="https://github.com/user-attachments/assets/93d5943a-27fe-4d0b-a483-b60ef89fc8c8" />

Reply:
Removed this added line as it doesn't solve the flakyness.